Frameworks/CoreFoundation/NumberDate.subproj/CFTimeZone.c

```diff
@@ -224,207 +191,48 @@ static CFMutableArrayRef __CFCopyRecursiveDirectoryList() {
-    for (idx = 0; idx < charcnt + 1; idx++) {
-    abbrs[idx] = NULL;
-    }
-    for (idx = 0; idx < cnt; idx++) {
-    CFAbsoluteTime at;
-    int32_t itime, offset;
-    uint8_t type, dst, abbridx;
-
-    at = (CFAbsoluteTime)(__CFDetzcode(timep) + 0.0) - kCFAbsoluteTimeIntervalSince1970;
-    if (0 == timecnt) itime = INT_MIN;
-    else if (at < (CFAbsoluteTime)INT_MIN) itime = INT_MIN;
-    else if ((CFAbsoluteTime)INT_MAX < at) itime = INT_MAX;
-    else itime = (int32_t)at;
-    timep += 4; /* harmless if 0 == timecnt */
-    type = (0 < timecnt) ? (uint8_t)*typep++ : 0;
-    if (typecnt <= type) {
-        result = false;
-        break;
-    }
-    offset = __CFDetzcode(ttisp + 6 * type);
-    dst = (uint8_t)*(ttisp + 6 * type + 4);
-    if (0 != dst && 1 != dst) {
-        result = false;
-        break;
-    }
-    abbridx = (uint8_t)*(ttisp + 6 * type + 5);
-    if (charcnt < abbridx) {
-        result = false;
-        break;
-    }
-    if (NULL == abbrs[abbridx]) {
-        abbrs[abbridx] = CFStringCreateWithCString(allocator, (char *)&charp[abbridx], kCFStringEncodingASCII);
-    }
-    __CFTZPeriodInit(*tzpp + idx, itime, abbrs[abbridx], offset, (dst ? true : false));
-    }
-    for (idx = 0; idx < charcnt + 1; idx++) {
-    if (NULL != abbrs[idx]) {
-        CFRelease(abbrs[idx]);
-    }
-    }
-    CFAllocatorDeallocate(allocator, abbrs);
-    if (result) {
-    // dump all but the last INT_MIN and the first INT_MAX
-    for (idx = 0; idx < cnt; idx++) {
-        if (((*tzpp + idx)->startSec == INT_MIN) && (idx + 1 < cnt) && (((*tzpp + idx + 1)->startSec == INT_MIN))) {
-        if (NULL != (*tzpp + idx)->abbrev) CFRelease((*tzpp + idx)->abbrev);
-        cnt--;
-        memmove((*tzpp + idx), (*tzpp + idx + 1), sizeof(CFTZPeriod) * (cnt - idx));
-        idx--;
-        }
-    }
-    // Don't combine these loops!  Watch the idx decrementing...
-    for (idx = 0; idx < cnt; idx++) {
-        if (((*tzpp + idx)->startSec == INT_MAX) && (0 < idx) && (((*tzpp + idx - 1)->startSec == INT_MAX))) {
-        if (NULL != (*tzpp + idx)->abbrev) CFRelease((*tzpp + idx)->abbrev);
-        cnt--;
-        memmove((*tzpp + idx), (*tzpp + idx + 1), sizeof(CFTZPeriod) * (cnt - idx));
-        idx--;
-        }
-    }
-    CFQSortArray(*tzpp, cnt, sizeof(CFTZPeriod), __CFCompareTZPeriods, NULL);
-    // if the first period is in DST and there is more than one period, drop it
-    if (1 < cnt && __CFTZPeriodIsDST(*tzpp + 0)) {
-        if (NULL != (*tzpp + 0)->abbrev) CFRelease((*tzpp + 0)->abbrev);
-        cnt--;
-        memmove((*tzpp + 0), (*tzpp + 0 + 1), sizeof(CFTZPeriod) * (cnt - 0));
-    }
-    *cntp = cnt;
-    } else {
-    CFAllocatorDeallocate(allocator, *tzpp);
-    *tzpp = NULL;
-    }
-    return result;
+CF_INLINE CFAbsoluteTime __UDateToCFAbsoluteTime(UDate time) {
+    // UDate = milliseconds since Jan 01, 1970
+    // CFAbsoluteTime = seconds since Jan 01, 2001
```

Comment:
might want to make these more full english sentences. I thought it was commented out code at first.